read_main_track.C

```diff
@@ -0,0 +1,191 @@
+//max_r and max_z are obtained from find_max_r_and_z.C
+#define max_r 156.763
+#define max_z 137.928
+
+R__LOAD_LIBRARY(/opt/WCSim/build/install/lib/libWCSimRoot.so)
+
+//iEvent is the event(in the given file) to be read
+int read_main_track(int iEvent=0, const char *filename="/work/kmtsui/wcte/cosmic/buggy/wcsim_mu-_0000.root") {
+
+    //Get file
+    TFile* file = new TFile(filename, "read");
+    TTree* tree = (TTree*)file->Get("wcsimT");
+    
+    //Check whether the given event exist
+    std::cout << "Target event: " << iEvent << "\n";
+    int nEvent = tree->GetEntries();    //number of event contain in the file
+    std::cout << "Total number of events: " << nEvent << "\n";
+    if (iEvent >= nEvent) {
+        std::cout << "Event " << iEvent << " does not exist\n";
+        return 1;
+    }
+
+    //Construct the event branch
+    WCSimRootEvent* wcsimrootsuperevent = new WCSimRootEvent();
+    tree->SetBranchAddress("wcsimrootevent",&wcsimrootsuperevent);
+    tree->GetBranch("wcsimrootevent")->SetAutoDelete(kTRUE);    //Idk what is this for but things seem to be ok without it.
```

Comment:
if you only read a single event, you don't need this line. However, when reading multiple event, this is needed to avoid memory leaks